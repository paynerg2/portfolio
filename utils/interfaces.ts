export interface Project {
    name: string;
    description: string;
    image: string;
    githubRepo: string;
    liveWebsite?: string;
    tags: string[];
}
