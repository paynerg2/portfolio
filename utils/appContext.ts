import React, { Dispatch, SetStateAction } from 'react';

interface IAppContext {
    selectedTags: string[];
    setSelectedTags: Dispatch<SetStateAction<string[]>>;
}

const AppContext = React.createContext<IAppContext>({} as IAppContext);

export default AppContext;
