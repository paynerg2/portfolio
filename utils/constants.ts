export const sectionNames = {
    projects: 'projects',
    skills: 'skills',
    about: 'about',
    contact: 'contact',
} as const;
