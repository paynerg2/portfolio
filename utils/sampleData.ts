import { Project } from './interfaces';
import TrackrPreview from '../assets/Trackr-Preview.png';
import MediaLibraryPreview from '../assets/Media-Library-Preview.png';
import CaveBarMenuPreview from '../assets/Cave_Bar_Menu_Preview.png';
import PortfolioPreview from '../assets/Portfolio_Preview.png';

export const projects: Project[] = [
    {
        name: 'Trackr',
        description:
            'Job hunting assistant for aspiring software developers. Tracks applications, interviews, and more. Automatically generates charts to visualize the process.',
        //@ts-ignore
        image: TrackrPreview,
        githubRepo: 'https://github.com/paynerg2/application-tracker',
        liveWebsite: 'https://devjob-trackr.com',
        tags: [
            'React',
            'Redux Toolkit',
            'Typescript',
            'Styled Components',
            'Framer Motion',
            'Nivo',
            'NodeJS',
            'Express',
            'MongoDB',
            'Jest',
            'Netlify',
            'Heroku',
            'C#',
            '.NET Core',
        ],
    },
    {
        name: 'Media Library',
        description:
            'Personal media library catalog. Allows users to manage their collection, and keep it available to quickly reference while on the hunt for new additions.',
        //@ts-ignore
        image: MediaLibraryPreview,
        githubRepo: 'https://github.com/paynerg2/Media-Library',
        liveWebsite: 'https://cave-media-library.herokuapp.com',
        tags: [
            'React',
            'Redux',
            'Typescript',
            'Styled Components',
            'React Spring',
            'NodeJS',
            'Express',
            'MongoDB',
            'Jest',
            'Enzyme',
            'Heroku',
        ],
    },
    {
        name: 'Cave Bar Menu',
        description:
            'An elegant menu commissioned for a home bar. Lets guests view available drinks, search by spirit, flavor profile, and more.',
        //@ts-ignore
        image: CaveBarMenuPreview,
        githubRepo: 'https://github.com/paynerg2/cave-bar-menu',
        liveWebsite: 'https://cave-bar-menu.herokuapp.com',
        tags: ['Javascript', 'GatsbyJS', 'CSS3', 'GraphQL', 'Contentful', 'Heroku'],
    },
    {
        name: 'Portfolio',
        description:
            'The portfolio you are currently viewing. Take a behind-the-scenes look at how it was made!',
        //@ts-ignore
        image: PortfolioPreview,
        githubRepo: 'https://github.com/paynerg2/portfolio',
        liveWebsite: 'https://www.rodneypayne.dev',
        tags: ['Typescript', 'NextJS', 'ChakraUI', 'Framer Motion', 'Jest', 'Vercel'],
    },
];
