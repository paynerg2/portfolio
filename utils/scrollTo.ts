interface Props {
    id: string;
    before?: () => void;
    after?: () => void;
}

export const scrollTo = ({ id, before, after }: Props) => {
    if (before !== undefined) {
        before();
    }

    // Some actions, such as closing the menu cancel the scroll animation if their animation has not finished beforehand
    // Small delay added to account for this
    setTimeout(() => {
        const section = document.querySelector(`#${id}`);
        if (section) {
            section.scrollIntoView({ behavior: 'smooth', block: 'start' });
        }
    }, 200);

    if (after !== undefined) {
        after();
    }
};
