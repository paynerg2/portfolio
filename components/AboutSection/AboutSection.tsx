import { Box, Divider, Heading, Stack, Text, VStack, Wrap } from '@chakra-ui/react';
import OutlineLabel from '../OutlineLabel/OutlineLabel';
import { icons } from '../../utils/skillIcons';
import { sectionNames } from '../../utils/constants';

const AboutSection = () => {
    return (
        <Stack id={sectionNames.about} direction="column" gap={6}>
            <Box alignSelf={{ base: 'center', md: 'flex-start' }}>
                <Heading>About Me</Heading>
                <Divider orientation="horizontal" w={100} ml={10} mb={5} h={1} />
            </Box>
            <Stack direction={{ base: 'column-reverse', md: 'row' }}>
                <Text
                    fontSize="xl"
                    textAlign={{ base: 'center', md: 'left' }}
                    flexGrow={1}
                    h="fit-content"
                    w={{ base: '100%', md: '50vw' }}
                    mt={{ base: 6, md: 0 }}
                >
                    My name is Rodney Payne. I&apos;ve always fostered a desire to learn,
                    understand, and engage with interesting and difficult topics. During my graduate
                    studies, I started writing programs to analyze data and control laboratory
                    systems, which led to a passion for building solutions with code.
                </Text>
                <VStack w={{ base: '100%', md: '50vw' }} h="fit-content" maxW="100%">
                    <Heading size="md">Appalachian State University</Heading>
                    <Divider orientation="horizontal" w={100} h={1} />
                    <Text color="blue.300">M.S. Engineering Physics</Text>
                    <Heading size="md">Appalachian State University</Heading>
                    <Divider orientation="horizontal" w={100} h={1} />
                    <Text color="blue.300">B.S. Mathematics, Applied Physics</Text>
                </VStack>
            </Stack>
            <Text fontSize="xl" textAlign={{ base: 'center', md: 'left' }}>
                Now, I&apos;m constantly dabbling with new technologies to find the best way to
                solve problems and keep up with the ever-changing landscape of software development.
            </Text>
            <OutlineLabel label="What I've worked with recently">
                <Wrap spacing={{ base: 1, md: 5 }}>
                    {(
                        [
                            'React',
                            'NextJS',
                            'Typescript',
                            'Javascript',
                            'Express',
                            'MongoDB',
                            'CSharp',
                        ] as (keyof typeof icons)[]
                    ).map((language) => icons[language])}
                </Wrap>
            </OutlineLabel>
        </Stack>
    );
};

export default AboutSection;
