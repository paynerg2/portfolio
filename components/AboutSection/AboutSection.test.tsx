import { render } from '../../test/test-utils';
import { sectionNames } from '../../utils/constants';
import AboutSection from './AboutSection';

describe('About Me Section', () => {
    it('renders the componenent', () => {
        const component = render(<AboutSection />);
        expect(component).toMatchSnapshot();
    });

    it('contains the corresponding section id', () => {
        const result = render(<AboutSection />);

        const element = result.container.querySelector(`#${sectionNames.about}`);
        expect(element).toBeInTheDocument();
    });
});
