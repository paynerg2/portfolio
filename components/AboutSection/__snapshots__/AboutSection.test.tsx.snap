// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About Me Section renders the componenent 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="chakra-ui-dark"
  >
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="chakra-stack css-c097x0"
        id="about"
      >
        <div
          class="css-1xpjxwp"
        >
          <h2
            class="chakra-heading css-1dklj6k"
          >
            About Me
          </h2>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-1xjzbn"
          />
        </div>
        <div
          class="chakra-stack css-1ub28e"
        >
          <p
            class="chakra-text css-udl9mk"
          >
            My name is Rodney Payne. I've always fostered a desire to learn, understand, and engage with interesting and difficult topics. During my graduate studies, I started writing programs to analyze data and control laboratory systems, which led to a passion for building solutions with code.
          </p>
          <div
            class="chakra-stack css-jvuvqr"
          >
            <h2
              class="chakra-heading css-18j379d"
            >
              Appalachian State University
            </h2>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-kb73j1"
            />
            <p
              class="chakra-text css-uqd5m9"
            >
              M.S. Engineering Physics
            </p>
            <h2
              class="chakra-heading css-18j379d"
            >
              Appalachian State University
            </h2>
            <hr
              aria-orientation="horizontal"
              class="chakra-divider css-kb73j1"
            />
            <p
              class="chakra-text css-uqd5m9"
            >
              B.S. Mathematics, Applied Physics
            </p>
          </div>
        </div>
        <p
          class="chakra-text css-1o97r86"
        >
          Now, I'm constantly dabbling with new technologies to find the best way to solve problems and keep up with the ever-changing landscape of software development.
        </p>
        <div
          class="css-1mfj9as"
        >
          <label
            aria-label="What I've worked with recently"
            class="css-1j5o14"
          >
            What I've worked with recently
          </label>
          <div
            class="chakra-wrap css-0"
          >
            <ul
              class="chakra-wrap__list css-1sypo4r"
            >
              <button
                class="chakra-button css-1g9agab"
                type="button"
              >
                <li
                  class="chakra-wrap__listitem css-1yp4ln"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="React"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </li>
              </button>
              <button
                class="chakra-button css-1g9agab"
                type="button"
              >
                <li
                  class="chakra-wrap__listitem css-1yp4ln"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="NextJS"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </li>
              </button>
              <button
                class="chakra-button css-1g9agab"
                type="button"
              >
                <li
                  class="chakra-wrap__listitem css-1yp4ln"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Typescript"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </li>
              </button>
              <button
                class="chakra-button css-1g9agab"
                type="button"
              >
                <li
                  class="chakra-wrap__listitem css-1yp4ln"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Javascript"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </li>
              </button>
              <button
                class="chakra-button css-1g9agab"
                type="button"
              >
                <li
                  class="chakra-wrap__listitem css-1yp4ln"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="Express"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </li>
              </button>
              <button
                class="chakra-button css-1g9agab"
                type="button"
              >
                <li
                  class="chakra-wrap__listitem css-1yp4ln"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="MongoDB"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </li>
              </button>
              <button
                class="chakra-button css-1g9agab"
                type="button"
              >
                <li
                  class="chakra-wrap__listitem css-1yp4ln"
                >
                  <span
                    style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                  >
                    <span
                      style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                    >
                      <img
                        alt=""
                        aria-hidden="true"
                        src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                        style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                      />
                    </span>
                    <img
                      alt="C#"
                      data-nimg="intrinsic"
                      decoding="async"
                      src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                      style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                    />
                    <noscript />
                  </span>
                </li>
              </button>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="chakra-stack css-c097x0"
      id="about"
    >
      <div
        class="css-1xpjxwp"
      >
        <h2
          class="chakra-heading css-1dklj6k"
        >
          About Me
        </h2>
        <hr
          aria-orientation="horizontal"
          class="chakra-divider css-1xjzbn"
        />
      </div>
      <div
        class="chakra-stack css-1ub28e"
      >
        <p
          class="chakra-text css-udl9mk"
        >
          My name is Rodney Payne. I've always fostered a desire to learn, understand, and engage with interesting and difficult topics. During my graduate studies, I started writing programs to analyze data and control laboratory systems, which led to a passion for building solutions with code.
        </p>
        <div
          class="chakra-stack css-jvuvqr"
        >
          <h2
            class="chakra-heading css-18j379d"
          >
            Appalachian State University
          </h2>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-kb73j1"
          />
          <p
            class="chakra-text css-uqd5m9"
          >
            M.S. Engineering Physics
          </p>
          <h2
            class="chakra-heading css-18j379d"
          >
            Appalachian State University
          </h2>
          <hr
            aria-orientation="horizontal"
            class="chakra-divider css-kb73j1"
          />
          <p
            class="chakra-text css-uqd5m9"
          >
            B.S. Mathematics, Applied Physics
          </p>
        </div>
      </div>
      <p
        class="chakra-text css-1o97r86"
      >
        Now, I'm constantly dabbling with new technologies to find the best way to solve problems and keep up with the ever-changing landscape of software development.
      </p>
      <div
        class="css-1mfj9as"
      >
        <label
          aria-label="What I've worked with recently"
          class="css-1j5o14"
        >
          What I've worked with recently
        </label>
        <div
          class="chakra-wrap css-0"
        >
          <ul
            class="chakra-wrap__list css-1sypo4r"
          >
            <button
              class="chakra-button css-1g9agab"
              type="button"
            >
              <li
                class="chakra-wrap__listitem css-1yp4ln"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="React"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </li>
            </button>
            <button
              class="chakra-button css-1g9agab"
              type="button"
            >
              <li
                class="chakra-wrap__listitem css-1yp4ln"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="NextJS"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </li>
            </button>
            <button
              class="chakra-button css-1g9agab"
              type="button"
            >
              <li
                class="chakra-wrap__listitem css-1yp4ln"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Typescript"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </li>
            </button>
            <button
              class="chakra-button css-1g9agab"
              type="button"
            >
              <li
                class="chakra-wrap__listitem css-1yp4ln"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Javascript"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </li>
            </button>
            <button
              class="chakra-button css-1g9agab"
              type="button"
            >
              <li
                class="chakra-wrap__listitem css-1yp4ln"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="Express"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </li>
            </button>
            <button
              class="chakra-button css-1g9agab"
              type="button"
            >
              <li
                class="chakra-wrap__listitem css-1yp4ln"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="MongoDB"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </li>
            </button>
            <button
              class="chakra-button css-1g9agab"
              type="button"
            >
              <li
                class="chakra-wrap__listitem css-1yp4ln"
              >
                <span
                  style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
                >
                  <span
                    style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
                  >
                    <img
                      alt=""
                      aria-hidden="true"
                      src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2780%27%20height=%2780%27/%3e"
                      style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
                    />
                  </span>
                  <img
                    alt="C#"
                    data-nimg="intrinsic"
                    decoding="async"
                    src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                    style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                  />
                  <noscript />
                </span>
              </li>
            </button>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
